js/modules/circles.js

```diff
@@ -77,13 +78,43 @@ function add_color_legend(scale_colors_fxn) {
       .attr("cx", function(d) { return d*circle_radius*2.2; })
       .attr("cy", function(d) { return 0; })
       .attr("r", circle_radius)
-      .attr("fill", function(d) { return scale_colors_fxn(d); })
+      .attr("fill", function(d) { return scale_colors_fxn.legend(d); })
+      .attr("stroke", "transparent")
       .on("mouseover", function(d) {
-        d3.select(this).attr("fill", "orange");
+        d3.select(this).attr("stroke", "orange").attr("stroke-width", 2);
+        var legend_color_str = scale_colors_fxn.legend(d);
+        d3.selectAll('.gage_point')
```

Comment:
Seems fine to me, but given there are a finite number of categories, a different approach that could have better performance could be to put all dots of a given category in a group, and then change the style on the group element, not on all of the individual ones. I have the sense that is faster, since you aren't searching 8K elements, you are searching 19 group elements. Just something to think about in the future, but not likely relevant since this will be replaced by canvas.

Reply:
Ahhh yes, that is a good idea. Thanks for mentioning it 👍